test/liveValidatorTests.ts

```diff
@@ -744,19 +744,84 @@ describe("Live Validator", () => {
         isPathCaseSensitive: false,
         useRelativeSourceLocationUrl: true,
         swaggerPathsPattern: [
-          "specification/signalr/resource-manager/Microsoft.SignalRService/**/*.json",
+          "specification/signalr/resource-manager/Microsoft.SignalRService/2021-01-01/*.json",
         ],
         git: {
           shouldClone: false,
         },
       };
       const liveValidator = new LiveValidator(options);
       await liveValidator.initialize();
-      const payload = require(`${__dirname}/liveValidation/payloads/xmsSecretAndPOST.json`);
+      const payload = require(`${__dirname}/liveValidation/payloads/xmsSecretAndPOST/xmsSecretAndPOST_2021-01-01.json`);
       const result = await liveValidator.validateLiveRequestResponse(payload);
       assert.equal(result.responseValidationResult.isSuccessful, true);
     });
 
+    it(`should not report error in response for POST,
+    when x-ms-secret is "true" and x-ms-mutability is "create" and "update"`, async () => {
+      const options = {
+        directory: `${__dirname}/liveValidation/swaggers/`,
+        isPathCaseSensitive: false,
+        useRelativeSourceLocationUrl: true,
+        swaggerPathsPattern: [
+          "specification/signalr/resource-manager/Microsoft.SignalRService/2021-02-01/*.json",
+        ],
+        git: {
+          shouldClone: false,
+        },
+      };
+      const liveValidator = new LiveValidator(options);
+      await liveValidator.initialize();
+      const payload = require(`${__dirname}/liveValidation/payloads/xmsSecretAndPOST/xmsSecretAndPOST_2021-02-01.json`);
+      const result = await liveValidator.validateLiveRequestResponse(payload);
+      assert.equal(result.responseValidationResult.isSuccessful, true);
+    });
+
+    it(`should report error in response for POST when x-ms-mutability is "create" and "update"`, async () => {
+      const options = {
+        directory: `${__dirname}/liveValidation/swaggers/`,
+        isPathCaseSensitive: false,
+        useRelativeSourceLocationUrl: true,
+        swaggerPathsPattern: [
+          "specification/signalr/resource-manager/Microsoft.SignalRService/2021-03-01/*.json",
+        ],
+        git: {
+          shouldClone: false,
+        },
+      };
+      const liveValidator = new LiveValidator(options);
+      await liveValidator.initialize();
+      const payload = require(`${__dirname}/liveValidation/payloads/xmsSecretAndPOST/xmsSecretAndPOST_2021-03-01.json`);
+      const result = await liveValidator.validateLiveRequestResponse(payload);
+      assert.equal(result.responseValidationResult.isSuccessful, false);
+    });
+
+    it(`should report error in response for httpMethod which is not POST,
+    when x-ms-secret is "true" or x-ms-mutability is "create" and "update"`, async () => {
+      const swaggers = [
```

Comment:
What's the difference between these three swaggers? can you add the main difference in the comment for better understanding?

Reply:
I've added comments in test
The general structure is as follows

- [√]httpMethod: POST, [√]x-ms-secret: "true", [X]x-ms-mutability: "create" and "update"
- [√]httpMethod: POST, [√]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"
- [√]httpMethod: POST, [X]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"
- [X]httpMethod: POST
  - [√]x-ms-secret: "true", [X]x-ms-mutability: "create" and "update"
  - [√]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"
  - [X]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"